vars/tdr.groovy

```diff
@@ -0,0 +1,11 @@
+def buildEndToEndTests(int delay, String buildJob, String[] params) {
+    build(job: buildJob, parameters: params, quietPeriod: delay, wait: false)
+}
+
+def setBuildStatus(String status, String desc) {
+    withCredentials([string(credentialsId: 'github-jenkins-api-key', variable: 'GITHUB_ACCESS_TOKEN')]) {
+        String sha = sh(script: 'git rev-parse HEAD', returnStdout: true).trim()
+        String url = "https://api.github.com/repos/nationalarchives/tdr-transfer-frontend/statuses/${sha}"
+        sh "curl -XPOST '${url}' -H 'Authorization: bearer ${env.GITHUB_ACCESS_TOKEN}' --data '{\"state\":\"${status}\",\"target_url\":\"${env.BUILD_URL}\",\"description\":\"${desc}\",\"context\":\"Jenkins TDR Front end test\"}'"
```

Comment:
The context is also hard-coded for the frontend app. What is the context used for, and is it OK to make it something generic like "TDR Jenkins build status"?

Reply:
The context is so that we could differentiate if there were other CI checks happening at the same time. Since we're not really doing that at the moment I can change it to something generic and we can change if we have more checks added in future.